compiler/src/dotty/tools/dotc/transform/Pickler.scala

```diff
@@ -50,13 +109,15 @@ object Pickler {
 class Pickler extends Phase {
   import ast.tpd.*
 
+  def doAsyncTasty(using Context): Boolean = ctx.asyncTastyPromise.isDefined
+
   override def phaseName: String = Pickler.name
 
   override def description: String = Pickler.description
 
   // No need to repickle trees coming from TASTY
   override def isRunnable(using Context): Boolean =
-    super.isRunnable && !ctx.settings.fromTasty.value
+    super.isRunnable && (!ctx.settings.fromTasty.value || doAsyncTasty)
```

Comment:
This change contradicts the comment on the method: "No need to repickle trees from TASTy".